### ISSUE TITLE
GCD is buggy for polynomial rings over number fields

### ISSUE THREAD

**Issue by HechtiDerLachs** [oscar-system/Oscar.jl#1886]
Try this:
```julia
T, t = PolynomialRing(QQ, "t");
K, a =  NumberField(2*t^2-1, "a");
R, (x, y) = PolynomialRing(K, ["x", "y"]);
x//y
```
For me, this produces the following error:
```
ERROR: Primitive element must be integral
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] __equation_order(K::AnticNumberField)
    @ Hecke ~/.julia/packages/Hecke/zEhcc/src/NumFieldOrd/NfOrd/NfOrd.jl:923
  [3] #1787
    @ ~/.julia/packages/Hecke/zEhcc/src/NumFieldOrd/NfOrd/NfOrd.jl:904 [inlined]
  [4] get!(default::Hecke.var"#1787#1788"{AnticNumberField}, h::Dict{Symbol, Any}, key::Symbol)
    @ Base ./dict.jl:465
  [5] get_attribute!
    @ ~/.julia/packages/AbstractAlgebra/Oe2Uj/src/Attributes.jl:225 [inlined]
  [6] EquationOrder
    @ ~/.julia/packages/Hecke/zEhcc/src/NumFieldOrd/NfOrd/NfOrd.jl:903 [inlined]
  [7] EquationOrder
    @ ~/.julia/packages/Hecke/zEhcc/src/NumFieldOrd/NfOrd/NfOrd.jl:902 [inlined]
  [8] equation_order
    @ ~/.julia/packages/Hecke/zEhcc/src/NumFieldOrd/NfRelOrd/NfRelOrd.jl:519 [inlined]
  [9] _gcd(f::AbstractAlgebra.Generic.MPoly{nf_elem}, g::AbstractAlgebra.Generic.MPoly{nf_elem}, ps::PrimesSet{Int64})
    @ Hecke.MPolyGcd ~/.julia/packages/Hecke/zEhcc/src/NumField/NfAbs/MPolyGcd.jl:273
 [10] gcd(f::AbstractAlgebra.Generic.MPoly{nf_elem}, g::AbstractAlgebra.Generic.MPoly{nf_elem})
    @ Hecke.MPolyGcd ~/.julia/packages/Hecke/zEhcc/src/NumField/NfAbs/MPolyGcd.jl:239
 [11] //(x::AbstractAlgebra.Generic.MPoly{nf_elem}, y::AbstractAlgebra.Generic.MPoly{nf_elem})
    @ AbstractAlgebra ~/.julia/packages/AbstractAlgebra/Oe2Uj/src/Fraction.jl:53
 [12] top-level scope
    @ REPL[4]:1
 [13] top-level scope
    @ ~/.julia/packages/Infiltrator/r3Hf5/src/Infiltrator.jl:710
```
As far as I can see, the `Frac`s try to call `gcd` on a pair of polynomials in `R` and that leads to a failure down the road. 

Expected behaviour: `x//y` returns an element of `FractionField(R)`.

(CC: @wdecker )

**Comment by thofma** [comment]
It is a problem in the GCD. Unfortunately at some places we assume that the defining polynomial of the number field is monic (and integral). @fieker: Should we just fall back to the 'generic' gcd if `is_defining_polynomial_nice(K)` fails?

**Comment by fieker** [comment]
should be fixed with the Hecke pr#953

**Comment by thofma** [comment]
Fixed once one does `Pkg.update()`.
